pkg/helpers/helpers.go

```diff
@@ -0,0 +1,26 @@
+// Copyright 2024 Nutanix. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+package helpers
+
+import (
+	"fmt"
+	"net/netip"
+)
+
+// IsIPInRange checks if the target IP falls within the start and end IP range (inclusive).
+func IsIPInRange(startIP, endIP, targetIP string) (bool, error) {
```

Comment:
What do you think about making this function more generic that can handle `-` ranges, `/` CIDRs, and a single IP.
This way it can be used here and in the other validation PR?
```suggestion
func IsIPInRange(ipRange, targetIP string) (bool, error) {
```

I think there is some useful code in https://github.com/nutanix-cloud-native/cloud-provider-nutanix/commit/5d9e97bc4b8b15d8f626345dcda57feea4f63a21#diff-0983f25d6bb690456a1aa84ca3f7f6e96981f1379a5fe09abced04dda99e8473R52-R71

Reply:
Makes sense but let's scope it out of this and have it in separate PR, it's been a while this tkt has been in review state.

Reply:
created https://jira.nutanix.com/browse/NCN-104787